---
bool isvalid
# the next available push plan if any are available
PushPlan plan
